### DISCUSSION TITLE
Renewal of subscriptions bought using Gift Cards

### DISCUSSION THREAD

**Discussion by igoraugustynski** [swellstores/community#398] [Help] [answered]
Hello,
as we migrated our customers towards the new shop which is using swell, some of them had annual subscriptions. As we moved towards swell and just monthly subscriptions, we created some gift cards for those users to allow them to use the new website without paying double. Those users bought subscriptions on our website using gift cards we provided. Today some of those subscriptions should renew, and what I expected was that the balance would be withdrawn from their gift cards. But I just saw one of those subscriptions (the first one I suppose) failing with the following error:
<img width="947" alt="Screenshot 2023-09-30 at 11 41 07" src="https://github.com/swellstores/community/assets/94485687/08d3f061-e490-4eb7-9e46-210c472a9006">

"Gift card not found" - even if the gift card does exist. **After** I saw this, I applied the gift card to the user account so that it would be converted to account balance - but even after that, if I press "retry" I get a "credit card not found" error. What is going on? How can I fix this?

**Comment by igoraugustynski** [comment]
<img width="1235" alt="Screenshot 2023-09-30 at 13 00 40" src="https://github.com/swellstores/community/assets/94485687/c99e2870-5b56-47ff-baed-6bad00514c24">
Another thing I have noticed is that some subscriptions are renewing silently - no invoices are generated and no transactions are made on the payment method. What is going on? This is unacceptable.

**Reply by ericingram** [reply]
👀

**Reply by igoraugustynski** [reply]
Please ask if You need additional information, I am very worried about this issue.

**Reply by igoraugustynski** [reply]
This particular issue (the one with the silent renewal) seems limited to the bundle we are selling as a subscription.

**Reply by igoraugustynski** [reply]
@ericingram maybe it has something to do with [Legacy Subscriptions](https://developers.swell.is/guides/core-concepts/purchase-options#updating-legacy-subscriptions)? Should i use a PUT request to set this products type to digital instead of bundle?

**Reply by ericingram** [reply]
I’m not sure, but we are investigating and will prioritize as necessary. I agree this is totally unacceptable.

**Reply by igoraugustynski** [reply]
Ok, thank You. We will also need to find a way to recover the missed payments after we fix the underlying issue. Is there a way to manually create the receipts and get the payments?

**Reply by igoraugustynski** [reply]
@ericingram Any news on this? We are very worried about this, our customers are basically using our product for free!

**Reply by igoraugustynski** [reply]
Guys, this is not an issue that can take 2 days to resolve. Please get back to me as soon as possible.

**Reply by steven-k1** [reply]
@igoraugustynski, Our engineers are currently investigating the issue. The time frame for resolving it ultimately depends on the complexity of the issue and the required fix. We understand the urgency of having a solution as soon as possible. We will keep you updated as updates become available.

**Reply by igoraugustynski** [reply]
In the meantime, how can we go about recovering the missed invoices? Is there a way to manually create invoices?

**Reply by steven-k1** [reply]
@igoraugustynski I'll need to confirm if this is possible, and I will get back to you with the details.

**Reply by igoraugustynski** [reply]
No news?

**Reply by igoraugustynski** [reply]
After some additional testing with new products I can confirm that the issue seems limited or at least somehow related to product bundles. Setting the bundle to set the type to 'digital' does not help in any way - the "First invoice" (note that it does not change to "Next invoice") date just skips to the next period end but the invoice is not created.

**Comment by steven-k1** [comment, marked as answer]
@igoraugustynski From the investigation of the issue, we were able to confirm the same details. The issue is related to bundles with digital products only. A fix for the issue is currently in QA, and once passed, we will have it deployed to production.

As for regenerating the missed invoices, we are confirming some additional details and should have an update on this shortly.

**Reply by igoraugustynski** [reply]
Hey Steven, that's good to hear. 
I think we can skip regenerating the invoices on swell -> We couldn't wait any longer so we are currently proceeding on manually creating the invoices and charging the cards using the Stripe dashboard.

**Reply by steven-k1** [reply]
@igoraugustynski Wanted to provide an update on the fix. 

Our QA team found some issues with the proposed fix, and additional changes were needed. The fix is being reviewed by QA once again, and we hope to have it ready for production shortly.

I'll follow up as I receive additional updates.

**Reply by steven-k1** [reply]
@igoraugustynski, The fix for the issue with generating bundle subscription invoices has been deployed to production.

Please review and confirm on your end. Let us know if you continue to encounter any issues.

**Reply by igoraugustynski** [reply]
Hey Steven, that's good. I see that we have some subscriptions renewing tomorrow, I'll see if they work correctly and let you know.

**Reply by igoraugustynski** [reply]
I can confirm that it seems to work now.

**Reply by steven-k1** [reply]
@igoraugustynski Glad to hear, and thanks for confirming.

The initial gift card issue is being looked into. I'll circle back when I have additional updates.
